src/tests/quantuminspire/qiskit/test_circuit_parser.py

```diff
@@ -130,24 +184,45 @@ def test_generate_cqasm_CorrectOutputGateU1(self):
         result = self._generate_cqasm_from_instructions(instructions, 3)
         self.assertTrue('Rz q[2], -0.785398\n' in result)
 
+        instructions[0]['conditional'] = {'mask': '0xF', 'type': 'equals', 'val': '0x3'}
+        result = self._generate_cqasm_from_instructions(instructions, 2)
+        self.assertTrue('not b[2,3]\nC-Rz b[0:3], q[2], -0.785398\nnot b[2,3]\n' in result)
+
         instructions = [{'name': 'u1', 'qubits': [2], 'params': [0.123456], 'texparams': ['0.123456']}]
         result = self._generate_cqasm_from_instructions(instructions, 3)
         self.assertTrue('Rz q[2], 0.123456\n' in result)
 
+        instructions[0]['conditional'] = {'mask': '0xF', 'type': 'equals', 'val': '0x3'}
+        result = self._generate_cqasm_from_instructions(instructions, 2)
+        self.assertTrue('not b[2,3]\nC-Rz b[0:3], q[2], 0.123456\nnot b[2,3]\n' in result)
+
         instructions = [{'name': 'u1', 'qubits': [0], 'params': [0], 'texparams': ['0']}]
         result = self._generate_cqasm_from_instructions(instructions, 2)
         self.assertFalse('q[0]' in result)
 
-    def test_generate_cqasm_CorrectOutputGateU2(self):
+        instructions[0]['conditional'] = {'mask': '0xF', 'type': 'equals', 'val': '0x3'}
+        result = self._generate_cqasm_from_instructions(instructions, 2)
+        self.assertFalse('q[0]' in result)
+
+    def test_generate_cqasm_correct_output_gate_u2(self):
         instructions = [{'name': 'u2', 'qubits': [0], 'params': [np.pi, np.pi / 2],
                          'texparams': ['\\pi', '\\frac{\\pi}{2}']}]
         result = self._generate_cqasm_from_instructions(instructions, 2)
         self.assertTrue('Rz q[0], 1.570796\nRy q[0], 1.570796\nRz q[0], 3.141593\n' in result)
 
+        instructions[0]['conditional'] = {'mask': '0xF', 'type': 'equals', 'val': '0x3'}
```

Comment:
Should the conditional gates not have their own test methods?

Reply:
Oke, I'll fix this. I agree it is better mainainable when they are tested in seperated tests.